bitcoin/src/amount.rs

```diff
@@ -1899,9 +1896,9 @@ mod tests {
         let p = Amount::from_str;
         let sp = SignedAmount::from_str;
 
-        assert_eq!(p("x BTC"), Err(E::InvalidCharacter('x')));
-        assert_eq!(p("5 BTC BTC"), Err(E::InvalidFormat));
-        assert_eq!(p("5 5 BTC"), Err(E::InvalidFormat));
+        assert_eq!(p("x BTC"), Err(E::UnknownDenomination("x BTC".into())));
```

Comment:
Would it make sense to add `rfind(' ')` in the error path to check for this? It'd make a nicer error message.

Reply:
To make the error messages a little better, I search for a space first, and only if I don't find one, search for the first alphabetic character.